internal/app/handlers/url.go

```diff
@@ -122,49 +154,230 @@ func (h *Handler) apiHandler(w http.ResponseWriter, r *http.Request) {
 		т.к. значение в apiHandlerRequest.URL будет пустое
 	*/
 	if err = json.Unmarshal(body, apiHandlerRequest); err != nil {
-		http.Error(w, err.Error(), 400)
+		http.Error(w, err.Error(), http.StatusBadRequest)
 		h.log.Errorf("cant't unmarshal request body, err: %s", err)
 
 		return
 	}
 
 	//генерируем ссылку и записываем её в хранилище
-	generatedURL := h.service.GenerateShortURL()
 	link := &models.Link{
-		ID:      generatedURL,
 		BaseURL: apiHandlerRequest.URL,
+		Hash:    cookieHash.Value,
 	}
-	err = h.service.Add(link)
+
+	var updated bool
+	updated, err = h.service.Add(ctx, link)
 	if err != nil {
-		http.Error(w, err.Error(), 400)
+		http.Error(w, err.Error(), http.StatusBadRequest)
 		h.log.Errorf("cant't add URL, err: %s", err)
 
 		return
 	}
 
 	//записываем результат в структуру ответа
-	apiHandlerResponse.Result = fmt.Sprintf("%s/%s", h.cfg.App.BaseURL, generatedURL)
+	apiHandlerResponse.Result = fmt.Sprintf("%s/%s", h.cfg.App.BaseURL, link.ID)
 
 	//записываем результат в виде json-объекта
 	result, err := json.Marshal(apiHandlerResponse)
 	if err != nil {
-		http.Error(w, err.Error(), 400)
+		http.Error(w, err.Error(), http.StatusBadRequest)
 		h.log.Errorf("cant't marshal result, err: %s", err)
 
 		return
 	}
 
 	//устанавливаем заголовок "application/json" и код ответа
 	w.Header().Set("Content-Type", "application/json; charset=utf-8")
-	w.WriteHeader(201)
+
+	responseStatusCode := http.StatusCreated
+
+	//меняем статус код в зафисимости от того добавили мы запись или проапдейтили
+	if updated {
+		responseStatusCode = http.StatusConflict
+	}
+	w.WriteHeader(responseStatusCode)
 
 	//записываем результат в тело ответа
 	_, err = w.Write(result)
 	if err != nil {
-		http.Error(w, err.Error(), 500)
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		h.log.Errorf("failed to write response body, err: %s", err)
+
+		return
+	}
+
+}
+
+func (h *Handler) apiGetAllURLS(w http.ResponseWriter, r *http.Request) {
+	ctx := context.Background()
+
+	cookieHash, err := r.Cookie("hash")
+
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadGateway)
+		h.log.Errorf("can't get hash from cookie, err: %s", err)
+
+		return
+	}
+
+	links, err := h.service.GetAll(ctx, cookieHash.Value)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusNoContent)
+		h.log.Info(err.Error())
+
+		return
+	}
+
+	var result []*APIGETAllResponse
```

Comment:
у тебя известна len(links), а размер result в точности равен размеру links, поэтому предлагаю 
```suggestion
	result:=make([]*APIGETAllResponse,0,len(links)
```